oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -508,6 +511,12 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(url, requestContent);
     adapter.initialize(request);
     request.setParser(parser);
+    request
+        .getHeaders()
+        .set(
+            MetricsUtils.API_CLIENT_HEADER,
+            MetricsUtils.getGoogleCredentialsMetricsHeader(
+                RequestType.ACCESS_TOKEN_REQUEST, getCredentialType()));
```

Comment:
nit: Maybe we can extract setting the metrics header out to a common helper function? If it makes sense, it could be something like:
```
MetricsUtils.setMetricHeader(request, {header-value})
```

Might be easier to add a test for. I'm also fine with this.